Box.V2/Request/HttpRequestHandler.cs

```diff
@@ -82,13 +84,9 @@ public class HttpRequestHandler : IRequestHandler
                         ((response.StatusCode == TooManyRequests && !isMultiPartRequest)
                         ||
                         (response.StatusCode == HttpStatusCode.Accepted && retryAfterHeader != null)) 
-                        && numRetries-- > 0)
+                        && retryCounter++ < 3)
```

Comment:
Can we pull this number into a constant at the top of the file and update its value to 5 to be in line with the other SDKs?  You should also update the test for the backoff counter to test values up to 5.